src/libbson/tests/test-utf8.c

```diff
@@ -81,6 +81,29 @@ test_bson_utf8_escape_for_json (void)
    str = bson_utf8_escape_for_json (unescaped, -1);
    BSON_ASSERT (0 == memcmp (str, "\\u000e", 7));
    bson_free (str);
+
+   // Invalid UTF-8 strings (should return null)
+
+   // 2 bytes expected
+   str = bson_utf8_escape_for_json ("\xc2", -1);
+   BSON_ASSERT (!str);
```

Comment:
```suggestion
   str = bson_utf8_escape_for_json ("\xc2", -1);
   BSON_ASSERT (!str);
   
   str = bson_utf8_escape_for_json ("...", -1);
   BSON_ASSERT (str);
   bson_free (str);
```
Include a test for the other side of the boundary condition.

Also applies to the other tests added below.